pkg/cmd/render/render.go

```diff
@@ -85,28 +113,44 @@ type TemplateData struct {
 // Run contains the logic of the render command.
 func (r *renderOpts) Run() error {
 	renderConfig := TemplateData{}
+
+	// XXX: What is this file used for?
 	if len(r.clusterConfigFile) > 0 {
 		_, err := ioutil.ReadFile(r.clusterConfigFile)
 		if err != nil {
 			return err
 		}
 		// TODO I'm thinking we parse this into a map and reference it that way
 	}
+
 	if err := r.manifest.ApplyTo(&renderConfig.ManifestConfig); err != nil {
 		return err
 	}
 	if err := r.generic.ApplyTo(
 		&renderConfig.FileConfig,
-
-		// TODO I don't think we have these
 		genericrenderoptions.Template{},
 		genericrenderoptions.Template{},
-
 		&renderConfig,
-		nil,
+		nil, // XXX: Can this be used to merge resources and write out config file? (map[string]resourcemerge.MergeFunc)
 	); err != nil {
 		return err
 	}
 
-	return genericrender.WriteFiles(&r.generic, &renderConfig.FileConfig, renderConfig)
+	if err := genericrender.WriteFiles(&r.generic, &renderConfig.FileConfig, renderConfig); err != nil {
+		return err
+	}
+
+	//if len(r.clusterInfrastructureInputFile) > 0 && (r.cloudProviderConfigInputFile) > 0 && len(r.cloudProviderConfigOutputFile) > 0 {
+
+	if len(r.clusterInfrastructureInputFile) > 0 && len(r.cloudProviderConfigOutputFile) > 0 {
```

Comment:
don't we already require that the input output file names be provided https://github.com/openshift/cluster-config-operator/pull/140/files#diff-5e45df1052a09901f6732e8a838541ebR85? so is this check required?

Reply:
Since the input and output file names are not required, the line referenced checks that if one is provided, the other must be provided as well. This line says that if they're both set, we can go ahead and call the transformer function.